course6/toyapp-adam/displaylib/surface_2d.cc

```diff
@@ -22,12 +24,35 @@ void Surface2D::Clear(Color color)
 {
 	for(int y = 0; y < mSurface->h; ++y)
 		for(int x = 0; x < mSurface->w; ++x)
-			SetPixel(Pixel{x, y}, color);
+		{
+			Pixel pixel{x, y};
+			SetPixel(pixel, color);
+			SetDepth(pixel, std::numeric_limits<float>::max());
+		}
+}
+
+void Surface2D::SetDepth(Pixel position, float depth)
+{
+	mDepthBuffer[position[1] * mSurface->w + position[0]] = depth;
```

Comment:
It looks to me like the DepthBuffer deserves to be a class by itself. You already have few methods for it, that are implemented in another object.